elastic/elastic-package
# [system testing] Add provision for system testing negative / false-positive scenarios in packages

This PR changes the way the `system testing` asserts at the number of hits in ElasticSearch when a system test is run.

Earlier, The testrunner exited when `enough` docs are found as per the config and there were more docs that got ingested after `assertHitCount` condition satisfied.

Now, The testrunner tries to get all the docs before checking for the `assertHitCount` and fails if the docs are more/less than the configured count

The `retryTicker` in the `waitUntilTrue` function is changed to `5 seconds` meaning , the retry interval for each poll to ES has been  changed to `5 seconds` and so if for `10 seconds`, there is no change in number of docs , the test passes.

### How to test

Run `PACKAGE_TEST_TYPE=false_positives PACKAGE_UNDER_TEST=httpjson_false_positive_asserts ./scripts/test-check-packages-false-positives.sh`

The `test-check-packages.sh` file has been modified to catch a failure from the new package `httpjson_false_positive_asserts` and a `testcase failure` is considered as `success scenario`

- Fixes #1308
